bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/discovery/MiIoDiscovery.java

```diff
@@ -66,11 +76,32 @@
 
     private final Logger logger = LoggerFactory.getLogger(MiIoDiscovery.class);
     private final CloudConnector cloudConnector;
+    private Map<String, String> cloudDevices = new ConcurrentHashMap<>();
+    private @Nullable Configuration miioConfig;
 
     @Activate
-    public MiIoDiscovery(@Reference CloudConnector cloudConnector) throws IllegalArgumentException {
+    public MiIoDiscovery(@Reference CloudConnector cloudConnector, @Reference ConfigurationAdmin configAdmin)
+            throws IllegalArgumentException {
         super(DISCOVERY_TIME);
         this.cloudConnector = cloudConnector;
+        try {
+            miioConfig = configAdmin.getConfiguration("binding.miio");
+        } catch (IOException | SecurityException e) {
+            logger.debug("Error getting configuration: {}", e.getMessage());
+        }
+    }
+
+    private String getCloudDiscoveryMode() {
+        if (miioConfig != null) {
+            try {
+                Dictionary<String, Object> properties = miioConfig.getProperties();
+                String cloudDiscoveryModeConfig = ((String) properties.get("cloudDiscoveryMode")).toLowerCase();
+                return Set.of(SUPPORTED, ALL).contains(cloudDiscoveryModeConfig) ? cloudDiscoveryModeConfig : DISABLED;
+            } catch (NullPointerException | ClassCastException | SecurityException e) {
```

Comment:
Please do proper null checks instead of just catching the NullPointerException.

Reply:
yes, did that... I did it in the former way to avoid one more dead code warning.
Somehow it assumes that the result of a Map.get is a notnull and complains about the null check.
I don't know how to tell it that it can be a Null